src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1468,6 +1472,15 @@ class HandshakeForDeflation : public HandshakeClosure {
   }
 };
 
+static size_t delete_monitors(GrowableArray<ObjectMonitor*> delete_list) {
+  size_t count = 0;
+  for (ObjectMonitor* monitor: delete_list) {
+    delete monitor;
```

Comment:
Just to be clear, this is the "delete monitor" call that is happening
while the calling JavaThread is in blocked state. If we did the release
of the oop in the ObjectMonitor in the ObjectMonitor's destructor, then
we would be calling release from blocked and not from thread-in-vm.

Reply:
This comment doesn't make sense to me. We are calling release whilst blocked IIUC.

Reply:
Ah that was the missing link. We delete deflated monitors and we release when deflating.